cloud.foundry.cli/src/main/java/cloud/foundry/cli/operations/ApplicationsOperations.java

```diff
@@ -47,68 +61,107 @@ public ApplicationsOperations(DefaultCloudFoundryOperations cloudFoundryOperatio
-     * The resulting mono will not perform any logging by default.
+     * Prepares a request for fetching applications data from the cloud foundry
+     * instance. The resulting mono will not perform any logging by default.
      *
-     * @return mono object of all applications as map of the application names as key and the ApplicationBeans as value
+     * @return mono object of all applications as map of the application names as
+     *         key and the ApplicationBeans as value
      */
     public Mono<Map<String, ApplicationBean>> getAll() {
+
         return this.cloudFoundryOperations
-                .applications()
-                .list()
-                .flatMap(this::getApplicationManifest)
-                .collectMap(ApplicationManifest::getName, ApplicationBean::new);
+            .applications()
+            .list()
+            .flatMap(applicationSummary -> Flux.zip(
+                getApplicationManifest(applicationSummary),
+                getMetadata(applicationSummary))
+                .collectMap(tuple -> tuple.getT1().getName(),
+                    tuple -> new ApplicationBean(tuple.getT1(), tuple.getT2())))
+            .reduce(new HashMap<String, ApplicationBean>(), (newMap, currentMap) -> {
+                newMap.putAll(currentMap);
+                return newMap;
+            });
     }
 
     private Mono<ApplicationManifest> getApplicationManifest(ApplicationSummary applicationSummary) {
         return this.cloudFoundryOperations
-                .applications()
-                .getApplicationManifest(GetApplicationManifestRequest
-                        .builder()
-                        .name(applicationSummary.getName())
-                        .build());
+            .applications()
+            .getApplicationManifest(GetApplicationManifestRequest
+                .builder()
+                .name(applicationSummary.getName())
+                .build());
+
+    }
+
+    private Mono<String> getMetadata(ApplicationSummary applicationSummary) {
+        GetApplicationRequest request = GetApplicationRequest.builder()
+            .applicationId(applicationSummary.getId())
+            .build();
+        return this.cloudFoundryOperations.getCloudFoundryClient()
+            .applicationsV3()
+            .get(request)
+            .flatMap(this::doGetMetadata);
+    }
 
+    private Mono<String> doGetMetadata(GetApplicationResponse getApplicationResponse) {
+     
+        List<String> metadata = new LinkedList<String>();
+        metadata.add(0, getApplicationResponse.getName());
+        Map<String, String> labels = getApplicationResponse.getMetadata().getLabels();
+        labels.keySet().forEach(action -> {
+            if (action.equals("version")) {
+                metadata.add(1, labels.get(action));
+            } else if (action.equals("branch")) {
+                metadata.add(2, labels.get(action));
+            }
+        });
```

Comment:
I think we agreed to store the entire metadata (including version, branch and so on) in a single label with a special constant key. This would have to be adjusted here.